channeldb/channel.go

```diff
@@ -3216,6 +3267,441 @@ const (
+		return err
+	}
+
+	if _, err = w.Write(b.Bytes()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeserializeChainActions reads out the chainAction map from the passed
+// reader.
+func DeserializeChainActions(r io.Reader) (map[string][]HTLC,
+	error) {
+
+	chainAction := make(map[string][]HTLC)
+
+	tlvStream, err := tlv.NewStream(
+		tlv.MakeDynamicRecord(
+			chainActionType, &chainAction,
+			HtlcActionRecordSize(&chainAction),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+	)
+
+	if err != nil {
+		return chainAction, err
+	}
+
+	var bodyLen int64
+	err = binary.Read(r, byteOrder, &bodyLen)
+	if err != nil {
+		return chainAction, err
+	}
+
+	lr := io.LimitReader(r, bodyLen)
+	if err = tlvStream.Decode(lr); err != nil {
+		return chainAction, err
+	}
+
+	return chainAction, nil
+}
+
+// serializeLocalForceCloseInsights writes out the passed
+// localForceCloseInsights to the writer.
+func serializeLocalForceCloseInsights(w io.Writer,
+	lc *LocalForceCloseInsights) error {
+
+	localForceCloseReasonByte := []byte(lc.Initiator)
+
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(localForceCloseInitiatorType,
+			&localForceCloseReasonByte),
+		tlv.MakeDynamicRecord(
+			chainActionType, &lc.ChainActions,
+			HtlcActionRecordSize(&lc.ChainActions),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+		tlv.MakePrimitiveRecord(broadCastHeightType,
+			&lc.BroadcastHeight),
+	)
+
+	if err != nil {
+		return err
+	}
+	var b bytes.Buffer
+	err = tlvStream.Encode(&b)
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(w, byteOrder, uint64(b.Len()))
+	if err != nil {
+		return err
+	}
+
+	if _, err = w.Write(b.Bytes()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deSerializeLocalForceCloseInsights reads out the LocalForceCloseInsights from
+// the passed reader.
+func deSerializeLocalForceCloseInsights(r io.Reader) (*LocalForceCloseInsights,
+	error) {
+
+	var (
+		initiator []byte
+		height    uint32
+	)
+
+	chainAction := make(map[string][]HTLC)
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(localForceCloseInitiatorType,
+			&initiator),
+		tlv.MakeDynamicRecord(
+			chainActionType, &chainAction,
+			HtlcActionRecordSize(&chainAction),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+		tlv.MakePrimitiveRecord(broadCastHeightType,
+			&height),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var bodyLen int64
+	err = binary.Read(r, byteOrder, &bodyLen)
+	if err != nil {
+		return nil, err
+	}
+
+	lr := io.LimitReader(r, bodyLen)
+	if err = tlvStream.Decode(lr); err != nil {
+		return nil, err
+	}
+
+	fcInsights := LocalForceCloseInsights{
+		Initiator:       LocalForceCloseInitiator(initiator),
+		ChainActions:    chainAction,
+		BroadcastHeight: height,
+	}
+
+	return &fcInsights, nil
+}
+
+// fetchLocalForceCloseInsights fetches the local force close insights
+// previously stored when commitment was broadcast.
+func fetchLocalForceCloseInsights(chanBucket kvdb.RBucket,
```

Comment:
This function should just return `(*LocalForceCloseInsights, error)` and not take the `*OpenChannel` argument. Favor returning things via expressions rather than mutating data structures.

Reply:
Looking at the code that calls this function, you would see this is how the functions look like, followed with the pattern for uniformity. I think though if there is an issue with it, it should be raised in a new issue:
https://github.com/lightningnetwork/lnd/blob/3906a44f17abfb4b7ffb3963de6515accec89017/channeldb/channel.go#L1856-L1886

Reply:
Oof. Yeah I see what you mean. I think this is a pretty bad pattern fwiw but I guess we can leave it as is. I'd request you rename it from `fetchLocal...` to `attachLocal...`